src/main/scala/battleship/routes/gameActor.scala

```diff
@@ -0,0 +1,84 @@
+package battleship.routes
+
+import akka.actor.{Actor, ActorLogging, ActorRef, Props}
+import battleship.PlayerActor.{GetNextShot, PlaceBoats}
+import battleship.game._
+import battleship.routes.gameActor.{StartGame, ThisIsBoatSetup, ThisIsNextMove}
+
+/**
+  * Created by m06f947 on 3-8-2017.
+  */
+
+
+
+object gameActor {
+
+  case class StartGame(boardsize:Int, player1:ActorRef, player2:ActorRef, boatset: Seq[Boat])
+
+  //case class WhatIsBoatSetup(boats:Seq[Boat], boardsize: Int)
+  case class ThisIsBoatSetup(placement: Set[(Boat,BoatLocation)])
+  //case class WhatIsNextMove(boardsize:Int, shotHistory: Seq[((Int,Int),ShotResult)])
+  case class ThisIsNextMove(x: Int, y: Int)
+
+
+  def props: Props = Props(new gameActor)
+
+}
+
+class gameActor extends Actor with ActorLogging {
+
+
+  //internal state
+  var boardsize = 8
+  var BoatSet = Seq(Boat(5), Boat(4), Boat(3), Boat(3), Boat(2))
+  //first of pair is currentplayer
+  var players: (ActorRef, ActorRef) = (Actor.noSender, Actor.noSender)
+  //Have the boatsetup ready. In here player1 is boatsetup._1
+  var boatsetup: Map[ActorRef, Set[(Boat, BoatLocation)]] = Map()
+  //keep the state of the boards
+  var boardstates: Map[ActorRef, BoardState] = Map()
+
+  def initPlayerState(playerToPlaceBoats: ActorRef): BoardState = {
+    val placements = boatsetup(playerToPlaceBoats)
+    placements.foldLeft(BoardState.empty) {
+      case (boardState, (boat, location)) => boardState.placeBoat(boat, location)
+    }
+  }
+
+  override def receive: Receive = {

```

Comment:
Omwille van de leesbaarheid zou je deze functie kunnen opsplitsen en de logica per ontvangen bericht in een aparte functie kunnen afhandelen.